commit cb7ab2eb9b3a669c72f29dabba4f5f60234a1ac7
Author: Ryan Tandy <rtandy@sd63.bc.ca>
Date:   Thu Feb 26 19:36:52 2015 +0000

    discard output from slaves when not in debug mode
    
    The demo slave generates some output via rfbLog. If no one reads the
    output channel, eventually the buffer fills and write() blocks, causing
    the demo server to hang. This fixes that by sending slave output to
    /dev/null when not forwarding it.

diff --git a/lib/src/Ipc/QtSlaveLauncher.cpp b/lib/src/Ipc/QtSlaveLauncher.cpp
index 9010c69..f32a11d 100644
--- a/lib/src/Ipc/QtSlaveLauncher.cpp
+++ b/lib/src/Ipc/QtSlaveLauncher.cpp
@@ -36,6 +36,9 @@
 
 #ifdef ITALC_BUILD_WIN32
 #include <windows.h>
+#define DEV_NULL "NUL"
+#else
+#define DEV_NULL "/dev/null"
 #endif
 
 namespace Ipc
@@ -64,11 +67,17 @@ void QtSlaveLauncher::start( const QStringList &arguments )
 	m_processMutex.lock();
 	m_process = new QProcess;
 
-	// forward stdout from slave to master when in debug mode
 	if( ItalcCore::config->logLevel() >= Logger::LogLevelDebug )
 	{
+		// forward stdout from slave to master when in debug mode
 		m_process->setProcessChannelMode( QProcess::ForwardedChannels );
 	}
+	else
+	{
+		// discard output when not in debug mode
+		m_process->setStandardOutputFile( DEV_NULL );
+		m_process->setStandardErrorFile( DEV_NULL );
+	}
 
 #ifndef DEBUG
 	m_process->start( applicationFilePath(), arguments );
